src/test/java/edu/cornell/kfs/fp/batch/service/impl/AdvanceDepositServiceImplTest.java

```diff
@@ -90,14 +93,7 @@ public void setUp() {
         mapping = new IncomingWireAchMapping();
         mapping.setShortDescription("ARMY");
 
-        ArrayList<String> methodNames = new ArrayList<>();
-        for (Method method : AdvanceDepositDocument.class.getMethods()) {
-            if (!Modifier.isFinal(method.getModifiers()) && !method.getName().startsWith("set") && !method.getName().startsWith("get")) {
-                methodNames.add(method.getName());
-            }
-        }
-        IMockBuilder<AdvanceDepositDocument> builder = EasyMock.createMockBuilder(AdvanceDepositDocument.class).addMockedMethods(methodNames.toArray(new String[0]));
-        advanceDepositDocument = builder.createNiceMock();
+        advanceDepositDocument = mock(AdvanceDepositDocument.class, CALLS_REAL_METHODS);
```

Comment:
using Mockito.mock with CALLS_REAL_METHODS, creates an unintiated object, and I think Mockito considers this unsafe.  Take a look at what I did with CuDisbursementVoucherDocumentTest.java.  You can user Powermockito to suppress a parent constructor, and then user power mockito to create a spy.

Reply:
thank you, I'll take a look at Powermockito and the example you pointed out.